crates/ruff_python_parser/src/lexer.rs

```diff
@@ -428,10 +429,46 @@ impl<'source> Lexer<'source> {
                     self.cursor.bump();
                     value.push('\\');
                 }
+                '?' => {
+                    self.cursor.bump();
+                    let mut question_count: u8 = 1;
+                    while self.cursor.eat_char('?') {
+                        question_count += 1;
+                    }
+
+                    // The help end magic command is valid if it has 1 or 2 question marks.
+                    // If the `value` is empty i.e., we are at the start of the line,
+                    // then we can't be sure if it is a help end magic command or not.
+                    if question_count > 2
+                        || previous_char_is_whitespace
+                        || value.is_empty()
```

Comment:
This is for the special case of having only question marks without any content (`????`). Help end magic command is only valid if it's preceded by non question mark / whitespace token.